### PR TITLE
docs(concepts): add explanation of Core/Universe from AI and Auto

### PR DESCRIPTION
**Pull request by mitsudome-r**
Signed-off-by: mitsudome-r <ryohsuke.mitsudome@tier4.jp>

## Description
Resolves https://github.com/autowarefoundation/autoware-documentation/issues/131.

Please let me know if there is a better place to put this document

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# How is Autoware Core/Universe Different from Autoware.AI and Autoware.Auto
```
Comment (LalithVipulananthan) [inline docs/design/autoware-concepts/difference-from-ai-and-auto.md:1]: Autoware docs use sentence case for titles [as recommended by the Google developer documentation style guide](https://developers.google.com/style/headings?hl=en)

```suggestion
# How is Autoware Core/Universe different from Autoware.AI and Autoware.Auto
```

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+## Autoware.Core/Universe
+
+In order to address issues we had in Autoware.Auto, we have decided to maintain code base with Autoware Core/Universe architecture. Autoware.Core would carry-over the policy of Autoware.Auto with a stable, well-tested code. In addition to that, we have created new repository Autoware.Universe which would act as an extension of Autoware.Core, where users can choose to replace a component (e.g. localization module) of Core with Universe component if they wish to use experimental features. Universe would have more relaxed code quality requirements compared to Core, but wouldn't be as loose as Autoware.AI. If we find features in Universe to be useful, we could also have AWF engineers porting those codes to be merged into Core as well. This way, we aim to achieve both requirements to have a stable and safe autonomous driving system while enabling access to state-of-the-art features generated by third-party contributors.
```
Comment (LalithVipulananthan) [inline docs/design/autoware-concepts/difference-from-ai-and-auto.md:22]: Changed `Autoware.Core` and `Autoware.Universe` to `Autoware Core` and `Autoware Universe` since those are supposed to be the official names of each codebase. If not, then we need to check the documentation site and AWF website for consistency.

I rewrote quite a lot of this section too, so again please check to make sure I have not removed or changed the originally intended meaning.

```suggestion
In order to address the issues with Autoware.Auto development, the Autoware Foundation decided to create a new architecture called Autoware Core/Universe.

Autoware Core carries over the original policy of Autoware.Auto to be a stable and well-tested codebase. Alongside Autoware Core is a new repository called Autoware Universe which acts as an extension of Autoware Core that allows users to replace a component (e.g. the Localization module) of Core with a Universe component in order to use more experimental features. 

Universe has a more relaxed code quality requirements compared to Core, but will not be as loose as that of Autoware.AI. Also, if any features added to Universe are deemed to be useful to the wider community, AWF engineers could potential port that code for merging into Core. 

This way, the primary requirement of having a stable and safe autonomous driving system can be achieved, whilst simultaneously enabling access to state-of-the-art features created by third-party contributors.
```

**Review by LalithVipulananthan — COMMENTED** [review]
As noted in the various comments, I have made some minor changes in places and then more significant changes in other places. Hopefully these will be of use. 😄

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+Autoware.Auto seemed to work fine initially, but after going under development of AVP and Cargo Delivery, we started to see some of the following issues:
+
+- The requirement set for merging features into Autoware.Auto is high, and it became difficult for researchers and students to join in to the development. Most of the developers of Autoware.Auto is from the companies who are in AWF organization. As a result, lacked place to have state-of-the-art features from papers.
+- What we also saw was that when we were trying to make a relatively large change to the code in architecture level, it was difficult to do on main branch of Autoware.Auto. To try out experimental architecture, we had very large overhead to keep main branch stable and also making sure every change satisfied the CI.
+
+## Autoware.Core/Universe
```
Comment (xmfcx) [inline docs/design/autoware-concepts/difference-from-ai-and-auto.md:20]:
```suggestion
## Autoware Core/Universe
```

Core and Universe aren't connected with a dot.

**Review by xmfcx — CHANGES_REQUESTED** [review]

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# How is Autoware Core/Universe Different from Autoware.AI and Autoware.Auto
+
+Autoware is the world's first "all-in-one" open-source software for self-driving vehicles. Since it was released in 2015, we had multiple releases with different concepts to improve our software.
+
+## Autoware.AI
+
+[Autoware.AI](https://github.com/Autoware-AI/autoware.ai) is the first distribution of Auto ｗ are that was released based on ROS1. The repository contains different packages to cover different aspects of autonomous driving technologies, including sensing, actuation, localization, mapping, perception, planning, and so on. While it successfully attracted many developers and had many contributions, we did not have particular use-cases or ODD defined for the software. It also lacked design documents to specify architecture, and each packages had different coding standards with very low test coverage. From the lessons learned from Autoware.AI development, we have decided to take different development process for Autoware.Auto to develop ROS2 version of Autoware.
```
[the remark this answers is not in the document]
Comment (kenji-miyake) [reply docs/design/autoware-concepts/difference-from-ai-and-auto.md:7]: I put some reference links here:
- https://developers.google.com/style/person?hl=en
- https://developers.google.com/style/voice?hl=en

**Comment by mitsudome-r** [comment]
Thanks @xmfcx @kenji-miyake @LalithVipulananthan.
I have applied the suggested changes.

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# How is Autoware Core/Universe different from Autoware.AI and Autoware.Auto?
+
+Autoware is the world's first "all-in-one" open-source software for self-driving vehicles.
+Since it was first released in 2015, there have been multiple releases made with differing underlying concepts, each one aimed at improving the software.
+
+## Autoware.AI
+
+[Autoware.AI](https://github.com/Autoware-AI/autoware.ai) is the first distribution of Autoware that was released based on ROS 1. The repository contains different packages to cover different aspects of autonomous driving technologies, including sensing, actuation, localization, mapping, perception, planning, and so on.
```
Comment (kenji-miyake) [inline docs/design/autoware-concepts/difference-from-ai-and-auto.md:8]: As a non-native speaker, I felt the words `different` are a bit unclear.
For me, this is easier to understand.

```suggestion
[Autoware.AI](https://github.com/Autoware-AI/autoware.ai) is the first distribution of Autoware that was released based on ROS 1. The repository contains a wide variety of packages to cover various aspects of autonomous driving technologies, including sensing, actuation, localization, mapping, perception, planning, and so on.
```

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+While it was successful in attracting many developers and contributions, there were no particular use-cases nor an ODD defined for the software. It also lacked design documents to specify architecture, and each package had different coding standards with very low test coverage.
```
Comment (kenji-miyake) [inline docs/design/autoware-concepts/difference-from-ai-and-auto.md:10]: I guess the part `there were no particular use-cases nor an ODD defined for the software` is a bit difficult to understand, but I have no alternative idea yet. :thinking:

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+From the lessons learned from Autoware.AI development, a different development process was taken for Autoware.Auto to develop a ROS 2 version of Autoware.
+
+!!! warning
+Autoware.AI is currently in maintenance mode and will reach end-of-life at the end of 2022.
```
Comment (kenji-miyake) [inline docs/design/autoware-concepts/difference-from-ai-and-auto.md:15]: You need an empty line.

```suggestion
!!! warning

    Autoware.AI is currently in maintenance mode and will reach end-of-life at the end of 2022.
```

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+## Autoware.Auto
+
+[Autoware.Auto](https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto) is an autonomous driving software based on ROS 2. As part of the transition to ROS 2, it was decided to avoid simply porting Autoware.AI from ROS 1 to ROS 2. Instead, the codebase was rewritten from scratch with proper engineering practices, including defining target use cases and ODDs (eg: Autonomous Valet Parking [AVP], Cargo Delivery, etc.), designing a proper architecture, writing design documents and test code.
```
Comment (kenji-miyake) [inline docs/design/autoware-concepts/difference-from-ai-and-auto.md:19]: [IMO]
- `an autonomous driving software` is obvious and redundant.
- Using the same format as Autoware.AI looks nice?

```suggestion
[Autoware.Auto](https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto) is the second distribution of Autoware that was released based on ROS 2. As part of the transition to ROS 2, it was decided to avoid simply porting Autoware.AI from ROS 1 to ROS 2. Instead, the codebase was rewritten from scratch with proper engineering practices, including defining target use cases and ODDs (eg: Autonomous Valet Parking [AVP], Cargo Delivery, etc.), designing a proper architecture, writing design documents and test code.
```

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+Autoware.Auto development seemed to work fine initially, but after completing the AVP and and Cargo Delivery ODD projects, we started to see the following issues:
+
+- The barrier to new engineers was too high
+  - A lot of work was required to merge new features into Autoware.Auto, and so it was difficult for researchers and students to contribute to development.
+  - As a consequence, most Autoware.Auto developers were from companies in the Autoware Foundation and so there were very few people who were able to add state-of-the-art features from research papers.
+- Making large scale architecture changes was too difficult
+  - To try out experimental architecture, there was a very large overhead involved in keeping the main branch stable whilst also making sure that every change satisfied the continuous integration requirements.
```
Comment (kenji-miyake) [inline docs/design/autoware-concepts/difference-from-ai-and-auto.md:27]: I guess we need full stops. @LalithVipulananthan Is that wrong?
https://developers.google.com/style/lists?hl=en

Also, fixed `large scale` to `large-scale` following Grammarly.

```suggestion
- The barrier to new engineers was too high.
  - A lot of work was required to merge new features into Autoware.Auto, and so it was difficult for researchers and students to contribute to development.
  - As a consequence, most Autoware.Auto developers were from companies in the Autoware Foundation and so there were very few people who were able to add state-of-the-art features from research papers.
- Making large-scale architecture changes was too difficult.
  - To try out experimental architecture, there was a very large overhead involved in keeping the main branch stable whilst also making sure that every change satisfied the continuous integration requirements.
```

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+## Autoware Core/Universe
+
+In order to address the issues with Autoware.Auto development, the Autoware Foundation decided to create a new architecture called Autoware Core/Universe.
+
+Autoware Core carries over the original policy of Autoware.Auto to be a stable and well-tested codebase. Alongside Autoware Core is a new repository called Autoware Universe which acts as an extension of Autoware Core that allows users to replace a component (e.g. the Localization module) of Core with a Universe component in order to use more experimental features.
```
Comment (kenji-miyake) [inline docs/design/autoware-concepts/difference-from-ai-and-auto.md:33]: Strictly speaking, I believe it's not a repository.

Also, I feel the sentence is a bit long.
https://developers.google.com/style/translation?hl=en#write-short,-clear,-and-precise-sentences

And regarding the word `experimental`, I think it's better to rephrase it to `advanced` or something like that.

```suggestion
Autoware Core carries over the original policy of Autoware.Auto to be a stable and well-tested codebase. Alongside Autoware Core is a new concept called Autoware Universe which acts as an extension of Autoware Core. Autoware Universe allows users to replace a component (e.g. the Localization module) of Core with a Universe component in order to use more advanced features.
```

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+Universe has a more relaxed code quality requirements compared to Core, but will not be as loose as that of Autoware.AI. Also, if any features added to Universe are deemed to be useful to the wider community, AWF engineers could potential port that code for merging into Core.
```
Comment (kenji-miyake) [inline docs/design/autoware-concepts/difference-from-ai-and-auto.md:35]: - First time to use `AWF`?
- The relationship between sentences is unclear to me.

Mm, I feel we can improve the second sentence, but I'm sorry I have no concrete idea. :pleading_face:

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+This way, the primary requirement of having a stable and safe autonomous driving system can be achieved, whilst simultaneously enabling access to state-of-the-art features created by third-party contributors.
```
Comment (kenji-miyake) [inline docs/design/autoware-concepts/difference-from-ai-and-auto.md:37]: Adding the link to https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-concepts/ at the end might be better.

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md (hunk 6 above)
Comment (LalithVipulananthan) [reply docs/design/autoware-concepts/difference-from-ai-and-auto.md:10]: We could explain what those terms mean but then this page will get very long! We could add a glossary of terms like `use-case` and `ODD` to the documentation or one of the wikis and then link to that?

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md (hunk 9 above)
Comment (LalithVipulananthan) [reply docs/design/autoware-concepts/difference-from-ai-and-auto.md:28]: I think full stops at the end of bullets varies depending on region. It's common in the US but not so common in the UK so full stops are something that I habitually leave out when writing bullet points 😅 . Since we are following the Google style guide for most cases, this is fine.

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md (hunk 5 above)
Comment (LalithVipulananthan) [reply docs/design/autoware-concepts/difference-from-ai-and-auto.md:8]: I note that the list seems to be all of things required for autonomous driving, so we could perhaps shorten the list to only 3-4 items or rephrase to say:

`The repository contains a wide variety of packages to cover the various aspects of autonomous driving technology - sensing, actuation, localization, mapping, perception and planning

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md (hunk 6 above)
Comment (kenji-miyake) [reply docs/design/autoware-concepts/difference-from-ai-and-auto.md:10]: I think we don't have the pages yet. :cry: 
Instead, I think it's good to add a supplemental explanation of what happened by "there were no particular use-cases nor an ODD defined for the software".
For example, the development efficiency was bad, etc.

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md (hunk 11 above)
Comment (LalithVipulananthan) [reply docs/design/autoware-concepts/difference-from-ai-and-auto.md:36]: So, we can either add `(AWF)` to the first instance of `Autoware Foundation`, or just change `AWF` here to `Autoware Foundation`. Honestly, since it only comes up three times, I think we can just expand out the acronym. 

As for the relationship between sentences, I believe they are simply just descriptions of the features of Autoware.Universe. Would it make more sense if the `Also` was removed and the second sentence started `If any features`? 🤔

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md (hunk 5 above)
Comment (LalithVipulananthan) [reply docs/design/autoware-concepts/difference-from-ai-and-auto.md:8]: My alternative suggestion:

```suggestion
[Autoware.AI](https://github.com/Autoware-AI/autoware.ai) is the first distribution of Autoware that was released based on ROS 1. The repository contains a variety of packages covering different aspects of autonomous driving technologies - sensing, actuation, localization, mapping, perception and planning.
```

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md (hunk 6 above)
Comment (LalithVipulananthan) [reply docs/design/autoware-concepts/difference-from-ai-and-auto.md:10]: How about this? It includes the reasoning for the Autoware architectural proposal from TIER IV, so should be good!

```suggestion
While it was successful in attracting many developers and contributions, it was difficult to improve Autoware.AI's capabilities for a number of reasons:

- A lack of concrete architecture design leading to a lot of built-up technical debt, such as tight coupling between modules and unclear module responsibility.
- Differing coding standards for each package, with very low test coverage.

Furthermore, there was no clear definition of the conditions under which an Autoware-enabled autonomous vehicle could operate, nor of the use cases or situations supported (eg: the ability to overtake a stationary vehicle). 
```

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md (hunk 5 above)
Comment (kenji-miyake) [reply docs/design/autoware-concepts/difference-from-ai-and-auto.md:8]: Cool! :+1:

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md (hunk 6 above)
Comment (kenji-miyake) [reply docs/design/autoware-concepts/difference-from-ai-and-auto.md:10]: So nice. :smile: Thank you!

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md (hunk 11 above)
Comment (LalithVipulananthan) [reply docs/design/autoware-concepts/difference-from-ai-and-auto.md:41]:
```suggestion
Universe has a more relaxed code quality requirements compared to Core, but will not be as loose as that of Autoware.AI. If any features added to Universe are deemed to be useful to the wider community, Autoware Foundation engineers could potentially port that code into Core.
```

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+- Making large-scale architecture changes was too difficult.
+  - To try out experimental architecture, there was a very large overhead involved in keeping the main branch stable whilst also making sure that every change satisfied the continuous integration requirements.
+
+## Autoware Core/Universe
+
+In order to address the issues with Autoware.Auto development, the Autoware Foundation decided to create a new architecture called Autoware Core/Universe.
+
+Autoware Core carries over the original policy of Autoware.Auto to be a stable and well-tested codebase. Alongside Autoware Core is a new concept called Autoware Universe which acts as an extension of Autoware Core. Autoware Universe allows users to replace a component (e.g. the Localization module) of Core with a Universe component in order to use more advanced features.
+
+Universe has a more relaxed code quality requirements compared to Core, but will not be as loose as that of Autoware.AI. Also, if any features added to Universe are deemed to be useful to the wider community, AWF engineers could potential port that code for merging into Core.
```
Comment (LalithVipulananthan) [inline docs/design/autoware-concepts/difference-from-ai-and-auto.md:41]: Formatting suggestion, plus some additional explanation about why code quality requirements are more relaxed compared to Core (this is my understanding, so please let me know if I have written anything that is incorrect).

```suggestion
Autoware Core carries over the original policy of Autoware.Auto to be a stable and well-tested codebase. Alongside Autoware Core is a new concept called Autoware Universe, which acts as an extension of Autoware Core with the following benefits:

- Users can easily replace a Core component with a Universe equivalent in order to use more advanced features, such as a new Localization or Perception algorithm. 
- Code quality requirements for Universe are more relaxed to make it easier for new developers, students and researchers to contribute, but will still be stricter than the requirements for Autoware.AI.
- Any advanced features added to Universe that are useful to the wider Autoware community will be reviewed and considered for potential inclusion in the main Autoware Core codebase.
```

File: docs/design/autoware-concepts/difference-from-ai-and-auto.md (hunk 12 above)
Comment (LalithVipulananthan) [reply docs/design/autoware-concepts/difference-from-ai-and-auto.md:43]: Something like this?

```suggestion
This way, the primary requirement of having a stable and safe autonomous driving system can be achieved, whilst simultaneously enabling access to state-of-the-art features created by third-party contributors. For more details about  the design of Autoware Core/Universe, refer to the [Autoware concepts documentation page]( https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-concepts/).
```
